code/modules/mob/dead/observer/freelook/marker/signal_abilities/psychic_tracer.dm

```diff
@@ -48,9 +48,19 @@
 
 
 /obj/effect/psychic_tracer/get_visualnet_tiles(var/datum/visualnet/network)
+
+	//Unsure why this is happening, but it seems that we aren't getting properly removed from chunk lists. Return PROCESS_KILL to ensure that happens
+	if (!EM)
+		if (!QDELETED(src))
```

Comment:
Why not a normal qdel()? That’s an extra timer needing to exist for each time this goes wrong, one to consider

Reply:
the thing is going to try to remove itself when deleted, and its going to be removed with returning this error code anyways, it felt like having all that in one frame might be redundant, and lead to possible race conditions or infinite loops

but you're right that this could cause lots of timers if there are many chunks holding onto one reference
what do you reckon, should i change it?